agents/gisfacisa.cc

```diff
@@ -0,0 +1,129 @@
+#include "gisfacisa.h"
+
+namespace cyclus {
+
+GISFacisa::GISFacisa(cyclus::Context* ctx)
+    : cyclus::Facility(ctx), capacity(100), cyclus::GIS(0, 0) {}
+
+GISFacisa::GISFacisa(cyclus::Context* ctx, float decimal_lat, float decimal_lon)
+    : cyclus::Facility(ctx),
+      capacity(100),
+      cyclus::GIS(decimal_lat, decimal_lon) {}
+
+std::string GISFacisa::str() {
+  // No info for now. Change later
+  return Facility::str();
+}
+
+std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
+GISFacisa::GetMatlRequests() {
+  using cyclus::CapacityConstraint;
+  using cyclus::Material;
+  using cyclus::RequestPortfolio;
+  using cyclus::Request;
+
+  std::set<RequestPortfolio<Material>::Ptr> ports;
+  RequestPortfolio<Material>::Ptr port(new RequestPortfolio<Material>());
+  double amt = Capacity();
+
+  Material::Ptr mat = cyclus::NewBlankMaterial(amt);
+  if (!recipe_name.empty()) {
+    Composition::Ptr c = context()->GetRecipe(recipe_name);
+    mat = Material::CreateUntracked(amt, c);
+  }
+
+  if (amt > cyclus::eps()) {
+    CapacityConstraint<Material> cc(amt);
+    port->AddConstraint(cc);
+
+    std::vector<std::string>::const_iterator it;
+    for (it = in_commods.begin(); it != in_commods.end(); ++it) {
+      port->AddRequest(mat, this, *it);
+    }
+
+    ports.insert(port);
+  }  // if amt > eps
+
+  return ports;
+}
+
+std::set<cyclus::RequestPortfolio<cyclus::Product>::Ptr>
+GISFacisa::GetProductRequests() {
+  using cyclus::CapacityConstraint;
+  using cyclus::Product;
+  using cyclus::RequestPortfolio;
+  using cyclus::Request;
+
+  std::set<RequestPortfolio<Product>::Ptr> ports;
+  RequestPortfolio<Product>::Ptr port(new RequestPortfolio<Product>());
+  double amt = Capacity();
+
+  if (amt > cyclus::eps()) {
+    CapacityConstraint<Product> cc(amt);
+    port->AddConstraint(cc);
+
+    std::vector<std::string>::const_iterator it;
+    for (it = in_commods.begin(); it != in_commods.end(); ++it) {
+      std::string quality = "";  // not clear what this should be..
```

Comment:
This is fine as is. It's a place to put recipe restrictions (e.g. only accept materials that have higher than 3% enriched uranium).